python/mxnet/optimizer.py

```diff
@@ -299,8 +300,13 @@ def update(self, index, weight, grad, state):
         if state:
             mom = state
             mom[:] *= self.momentum
-            mom[:] += -lr * (grad + wd * weight)
-            weight[:] += mom
+            grad += wd * weight
```

Comment:
@mavenlin Let us do this. Create a new NAGOptimizer to subclass SGDOptimizer and add your stuffs there